nlp-engine/app_vf.py

```diff
@@ -0,0 +1,181 @@
+    #src1=inp.src
+    #inp=inp.text
+    inp=inp.lower()
+    
+
+    url = "https://api.gupshup.io/sm/api/v1/msg"
+    headers = {
+      'Cache-Control': 'no-cache',
+      'Content-Type': 'application/x-www-form-urlencoded',
+      'apikey': '37cef3c8bf164df7cdc0a36eae94beec',
+      'cache-control': 'no-cache'
+                      }
+
+    if len(inp.split())==1 and inp!="hi" and inp!="hello":
+        answer=find_city(inp)[0].upper()
+        answer=answer[0]+answer[1:].lower()
+
+        k=payload.index("&message")
+        payload=payload[0:k]+"&message="+translator.translate("We have detected *"+answer+"* as your city.",dest='en').text+ "&src.name=chatbotpayment"
+        response = requests.request("POST", url, headers=headers, data = payload)
+
+        payload=payload[0:k]+"&message="+translator.translate("If your city is detected correctly, you can continue with your bill/receipt queries.\n Else, kindly type your city name again.\n ",dest='en').text+ "&src.name=chatbotpayment"
+        response = requests.request("POST", url, headers=headers, data = payload)
+        return ""
+        
+
+    inp=translator.translate(inp,dest='en')
+    src1=inp.src
+    inp=inp.text
+    res= process(inp)
+    
+    c=res.split()
+
+    if c[0]=="Welcome" :
+        
+        k=payload.index("&message")
+        payload=payload[0:k]+"&message="+translator.translate(res,dest=src1).text+ "&src.name=chatbotpayment"
+        response = requests.request("POST", url, headers=headers, data = payload)
+        
+        
+        k=payload.index("&message")
+        payload=payload[0:k]+"&message="+translator.translate("Please enter your city name\n",dest=src1).text+ "&src.name=chatbotpayment"
+        #response = requests.request("POST", url, headers=headers, data = payload)
+
+        
+            
+        
+        
+    elif c[0]=='Showing':
+        if 'water' in c:
+            payload = 'channel=whatsapp&source=917834811114&destination='+destination+'&message=%7B%22type%22%3A%22file%22%2C%22url%22%3A%22https%3A//www.buildquickbots.com/whatsapp/media/sample/pdf/sample01.pdf%22%2C%22caption%22%3A%22%22%2C%22filename%22%3A%22Water_Sewerage_receipts.pdf%22%7D&src.name=chatbotpayment'
+
+        elif 'trade' in c:
+            payload = 'channel=whatsapp&source=917834811114&destination='+destination+'&message=%7B%22type%22%3A%22file%22%2C%22url%22%3A%22https%3A//www.buildquickbots.com/whatsapp/media/sample/pdf/sample01.pdf%22%2C%22caption%22%3A%22%22%2C%22filename%22%3A%22Trade_license_receipts.pdf%22%7D&src.name=chatbotpayment'
+
+        elif 'property' in c:
+            payload = 'channel=whatsapp&source=917834811114&destination='+destination+'&message=%7B%22type%22%3A%22file%22%2C%22url%22%3A%22https%3A//www.buildquickbots.com/whatsapp/media/sample/pdf/sample01.pdf%22%2C%22caption%22%3A%22%22%2C%22filename%22%3A%22Property_Tax_receipts.pdf%22%7D&src.name=chatbotpayment'
+
+        else:
+            
+            k=payload.index("&message")

```

Comment:
add comments in code to explain what is happening in the code

Reply:
Done